test/multiverse/suites/stripe/stripe_instrumentation_test.rb

```diff
@@ -0,0 +1,160 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'json'
+require 'stripe'
+require 'net/http'
+
+class StripeInstrumentation < Minitest::Test
+  API_KEY = '123456789'
+
+  def setup
+    Stripe.api_key = API_KEY
+    # Creating a new connection and response, which both get stubbed
+    # later, helps us get around needing to provide a valid API key
+    @connection = Stripe::ConnectionManager.new
+    @response = Net::HTTPResponse.new('1.1', '200', 'OK')
+    # Bypass #stream_check ("attempt to read body out of block")
+    @response.instance_variable_set(:@read, true)
+    @response.body = {
+      object: 'list',
+      data: [{'id': '12134'}],
+      has_more: false,
+      url: '/v1/charges'
+    }.to_json
+  end
+
+  def test_version_supported
+    assert(Stripe::VERSION >= '5.38.0')
+  end
+
+  def test_subscribed_request_begin
+    subcribers = Stripe::Instrumentation.send(:subscribers)
+    newrelic_begin_subscriber = subcribers[:request_begin].detect { |_k, v| v.to_s.include?('instrumentation/stripe') }
+
+    assert(newrelic_begin_subscriber)
+  end
+
+  def test_subscribed_request_end
+    subcribers = Stripe::Instrumentation.send(:subscribers)
+    newrelic_begin_subscriber = subcribers[:request_end].detect { |_k, v| v.to_s.include?('instrumentation/stripe') }
+
+    assert(newrelic_begin_subscriber)
+  end
+
+  def test_newrelic_segment
+    Stripe::StripeClient.stub(:default_connection_manager, @connection) do
```

Comment:
Seeing as you use this wrapper multiple times, it'd be a good candidate for a test helper.

```ruby
def with_stubbed_connection_manager(&block)
  Stripe::StripeClient.stub(:default_connection_manager, @connection) do
    @connection.stub(:execute_request, @response) do
      block.call
    end
  end
end
```

then for each test you could use it:

```ruby
with_stubbed_connection_manager do
  in_transaction do |txn|
    Stripe::Customer.list({limit: 3})
    stripe_segment = txn.segments.detect { |s| s.name == 'Stripe/v1/customers get' }

    assert(stripe_segment)
  end
end
```

Reply:
Pattern into a method.. love and done! https://github.com/newrelic/newrelic-ruby-agent/pull/2180/commits/63014dd5d7236c3fb38390840d7cfc0ff0ed70a4